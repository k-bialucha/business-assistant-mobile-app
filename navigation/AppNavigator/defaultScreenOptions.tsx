import { Platform } from 'react-native';

import { StackNavigationOptions } from '@react-navigation/stack';

import Colors from '../../theme/Colors';

const defaultScreenOptions: StackNavigationOptions = {
  headerStyle: {
    backgroundColor: Platform.OS === 'android' ? Colors.gray : 'white',
  },
  headerTintColor: Platform.OS === 'android' ? 'white' : Colors.gray,
};

export default defaultScreenOptions;
