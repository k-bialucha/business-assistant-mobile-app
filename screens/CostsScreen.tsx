import React from 'react';
import { View, Text } from 'react-native';

const CostsScreen = () => {
  return (
    <View>
      <Text>Costs</Text>
    </View>
  );
};

export default CostsScreen;
