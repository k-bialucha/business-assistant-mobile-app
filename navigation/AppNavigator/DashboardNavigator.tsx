import React from 'react';
import {
  createStackNavigator,
  StackNavigationProp,
} from '@react-navigation/stack';
import { RouteProp } from '@react-navigation/native';

import { RootStackParamList } from '.';
import defaultScreenOptions from './defaultScreenOptions';

import DashboardScreen from '../../screens/DashboardScreen';

const DashboardStackNavigator = createStackNavigator<RootStackParamList>();

type ScreenNavigationProp = StackNavigationProp<
  RootStackParamList,
  'Dashboard'
>;
type ScreenRouteProp = RouteProp<RootStackParamList, 'Dashboard'>;

type DashboardNavigationProps = React.FC<{
  navigation: ScreenNavigationProp;
  route: ScreenRouteProp;
}>;

const DashboardNavigator: DashboardNavigationProps = () => {
  return (
    <DashboardStackNavigator.Navigator screenOptions={defaultScreenOptions}>
      <DashboardStackNavigator.Screen
        name="Dashboard"
        component={DashboardScreen}
      />
    </DashboardStackNavigator.Navigator>
  );
};

export default DashboardNavigator;
