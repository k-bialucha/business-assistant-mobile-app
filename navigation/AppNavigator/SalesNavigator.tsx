import React from 'react';

import { RouteProp } from '@react-navigation/native';
import {
  createStackNavigator,
  StackNavigationProp,
} from '@react-navigation/stack';

import SalesEntryScreen from '../../screens/SalesEntryScreen';
import SalesScreen, { SalesScreenNavOptions } from '../../screens/SalesScreen';

import { ParamList } from '.';
import defaultScreenOptions from './defaultScreenOptions';

const SalesStackNavigator = createStackNavigator();

type ScreenNavigationProp = StackNavigationProp<ParamList, 'Sales'>;

type ScreenRouteProp = RouteProp<ParamList, 'Sales'>;

type SalesNavigationProps = React.FC<{
  navigation: ScreenNavigationProp;
  route: ScreenRouteProp;
}>;

const SalesNavigator: SalesNavigationProps = () => {
  return (
    <SalesStackNavigator.Navigator screenOptions={defaultScreenOptions}>
      <SalesStackNavigator.Screen
        name="SalesList"
        component={SalesScreen}
        options={SalesScreenNavOptions}
      />
      <SalesStackNavigator.Screen
        name="SalesEntry"
        component={SalesEntryScreen}
      />
    </SalesStackNavigator.Navigator>
  );
};

export default SalesNavigator;
