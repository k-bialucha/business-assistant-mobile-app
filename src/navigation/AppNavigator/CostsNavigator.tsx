import React from 'react';

import { CompositeNavigationProp, RouteProp } from '@react-navigation/native';
import {
  createStackNavigator,
  StackNavigationProp,
} from '@react-navigation/stack';

import CostsEntryScreen from '~/screens/CostsEntryScreen';
import CostsListScreen, {
  CostsListScreenNavOptions,
} from '~/screens/CostsListScreen';

import defaultScreenOptions from './defaultScreenOptions';
import {
  AppNavigatorNavigationProp,
  NavigationData as AppNavigatorNavigationData,
} from './index';

type ParamList = {
  CostsList: undefined;
  CostsEntry: undefined;
};

const CostsStackNavigator = createStackNavigator<ParamList>();

type Props = AppNavigatorNavigationData<'Costs'>;

const CostsNavigator: React.FC<Props> = () => {
  return (
    <CostsStackNavigator.Navigator screenOptions={defaultScreenOptions}>
      <CostsStackNavigator.Screen
        name="CostsList"
        component={CostsListScreen}
        options={CostsListScreenNavOptions}
      />
      <CostsStackNavigator.Screen
        name="CostsEntry"
        component={CostsEntryScreen}
      />
    </CostsStackNavigator.Navigator>
  );
};

export default CostsNavigator;

// combine navigation prop for nested navigators
// (AppNavigator data + this StackNavigator data)
export type NavigationPropCombined<
  RouteName extends keyof ParamList
> = CompositeNavigationProp<
  AppNavigatorNavigationProp<'Costs'>,
  StackNavigationProp<ParamList, RouteName>
>;

// export type so screens can get typing
export type NavigationData<RouteName extends keyof ParamList> = {
  navigation: NavigationPropCombined<RouteName>;
  route: RouteProp<ParamList, RouteName>;
};
