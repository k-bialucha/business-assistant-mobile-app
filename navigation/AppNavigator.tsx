import React from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { createStackNavigator } from '@react-navigation/stack';

import DashboardScreen from '../screens/DashboardScreen';
import SalesScreen from '../screens/SalesScreen';
import CostsScreen from '../screens/CostsScreen';
import CostsEntryScreen from '../screens/CostsEntryScreen';
import SalesEntryScreen from '../screens/SalesEntryScreen';

const DashboardStackNavigator = createStackNavigator();

const DashboardNavigator = () => {
  return (
    <DashboardStackNavigator.Navigator>
      <DashboardStackNavigator.Screen
        name="Dashboard"
        component={DashboardScreen}
      />
    </DashboardStackNavigator.Navigator>
  );
};

const SalesStackNavigator = createStackNavigator();

const SalesNavigator = () => {
  return (
    <SalesStackNavigator.Navigator>
      <SalesStackNavigator.Screen name="SalesList" component={SalesScreen} />
      <SalesStackNavigator.Screen
        name="SalesEntry"
        component={SalesEntryScreen}
      />
    </SalesStackNavigator.Navigator>
  );
};

const CostsStackNavigator = createStackNavigator();

const CostsNavigator = () => {
  return (
    <CostsStackNavigator.Navigator>
      <CostsStackNavigator.Screen name="CostsList" component={CostsScreen} />
      <CostsStackNavigator.Screen
        name="CostsEntry"
        component={CostsEntryScreen}
      />
    </CostsStackNavigator.Navigator>
  );
};

const AppTabNavigator = createBottomTabNavigator();

const AppNavigator = () => {
  return (
    <AppTabNavigator.Navigator>
      <AppTabNavigator.Screen name="Dashboard" component={DashboardNavigator} />
      <AppTabNavigator.Screen name="Sales" component={SalesNavigator} />
      <AppTabNavigator.Screen name="Costs" component={CostsNavigator} />
    </AppTabNavigator.Navigator>
  );
};

export default AppNavigator;
