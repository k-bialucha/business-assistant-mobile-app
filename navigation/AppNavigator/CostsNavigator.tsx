import React from 'react';

import { createStackNavigator } from '@react-navigation/stack';

import CostsEntryScreen from '../../screens/CostsEntryScreen';
import CostsScreen, { CostsScreenNavOptions } from '../../screens/CostsScreen';

import defaultScreenOptions from './defaultScreenOptions';
import { ScreenProps } from './index';

const CostsStackNavigator = createStackNavigator();

type Props = ScreenProps<'Costs'>;

const CostsNavigator: React.FC<Props> = () => {
  return (
    <CostsStackNavigator.Navigator screenOptions={defaultScreenOptions}>
      <CostsStackNavigator.Screen
        name="CostsList"
        component={CostsScreen}
        options={CostsScreenNavOptions}
      />
      <CostsStackNavigator.Screen
        name="CostsEntry"
        component={CostsEntryScreen}
      />
    </CostsStackNavigator.Navigator>
  );
};

export default CostsNavigator;
