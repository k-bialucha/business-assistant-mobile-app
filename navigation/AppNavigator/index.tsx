import React from 'react';
import { Platform } from 'react-native';

import { Ionicons } from '@expo/vector-icons';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { RouteProp } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';

import Colors from '../../theme/Colors';

import CostsNavigator from './CostsNavigator';
import DashboardNavigator from './DashboardNavigator';
import SalesNavigator from './SalesNavigator';
import SettingsNavigator from './SettingsNavigator';

export type RootStackParamList = {
  Dashboard: undefined;
  Sales: undefined;
  Costs: undefined;
  Settings: undefined;
};

export type ScreenProps<RouteName extends keyof RootStackParamList> = {
  navigation: StackNavigationProp<RootStackParamList, RouteName>;
  route: RouteProp<RootStackParamList, RouteName>;
};

const AppTabNavigator = createBottomTabNavigator<RootStackParamList>();

const AppNavigator: React.FC<{}> = () => {
  return (
    <AppTabNavigator.Navigator
      initialRouteName="Dashboard"
      screenOptions={({ route }) => ({
        tabBarIcon: ({ color, size }) => {
          const icons = {
            Dashboard: Platform.OS === 'android' ? 'md-stats' : 'ios-stats',
            Sales:
              Platform.OS === 'android' ? 'md-trending-up' : 'ios-trending-up',
            Costs:
              Platform.OS === 'android'
                ? 'md-trending-down'
                : 'ios-trending-down',
            Settings:
              Platform.OS === 'android' ? 'md-settings' : 'ios-settings',
          };

          return (
            <Ionicons name={icons[route.name]} size={size} color={color} />
          );
        },
      })}
      tabBarOptions={{
        showLabel: false,
        activeTintColor: Colors.navyBlue,
        inactiveTintColor: Colors.silver,
      }}
    >
      <AppTabNavigator.Screen name="Sales" component={SalesNavigator} />
      <AppTabNavigator.Screen name="Dashboard" component={DashboardNavigator} />
      <AppTabNavigator.Screen name="Costs" component={CostsNavigator} />
      <AppTabNavigator.Screen name="Settings" component={SettingsNavigator} />
    </AppTabNavigator.Navigator>
  );
};

export default AppNavigator;
