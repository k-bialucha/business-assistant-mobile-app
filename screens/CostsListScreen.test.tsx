import React from 'react';

import { shallow } from 'enzyme';

import { NavigationData } from '../navigation/AppNavigator/CostsNavigator';

import CostsScreen from './CostsScreen';

type Props = NavigationData<'CostsList'>;

const fakeProps: Props = {
  route: { key: '1234', name: 'CostsList' },
  // @ts-ignore
  navigation: {},
};

describe('<CostsScreen />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<CostsScreen {...fakeProps} />);

    expect(tree).toMatchSnapshot();
  });
});
