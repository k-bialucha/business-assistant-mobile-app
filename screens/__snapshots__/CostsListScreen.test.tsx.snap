// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CostsScreen /> matches the snapshot 1`] = `
<View>
  <Text>
    Costs
  </Text>
</View>
`;
